bootstrap/src/contexts/Auth.js

```diff
@@ -21,51 +21,52 @@ const getUser = async () => {
 export const AuthProvider = ({ children }) => {
     const [userState, setUserState] = useState({ ...defaultSourceState });
     useEffect(() => {
-        getUser()
-            .then(account => {
-                setUserState({ isLoading: false, error: null, data: account });
-            })
-            .catch(error => {
+        const initializeUser = async () => {
+            try {
+                const user = await getUser();
+                setUserState({ isLoading: false, error: null, data: user });
+            } catch (error) {
                 setUserState({ isLoading: false, error, data: null });
-            });
+            }
+        };
+        initializeUser();
     }, []);
 
     const login = useCallback(async (username, password) => {
-        setUserState({ isLoading: true, error: null, data: null });
-        return sdk.user
-            .login(username, password)
-            .then(user => {
-                setUserState({ isLoading: false, error: null, data: user });
-                return user;
-            })
-            .catch(error => {
-                setUserState({ isLoading: false, error, data: null });
-                throw new ApiResponseError(error.message, error.response, error.responseBody);
-            });
+        try {
+            setUserState({ isLoading: true, error: null, data: null });
+            await sdk.user.login(username, password);
```

Comment:
`sdk.user.login` should already return the `user`, so there is no need to call `getUser` (cf. the original code).

Reply:
unfortunatelly `sdk.user.login` and `getUser` return objects with completely different structure. `getUser` has to be used here to keep consistency with user state update on line 27.

Reply:
Oh, I see, let's leave it then, thanks for the clarification :)